### PR TITLE
PLANNER-1709 Avoid deprecated penalize/reward overloads

### PR DESCRIPTION
**Pull request by triceo**
**Check list**:

Your pull request:

- [x] targets the `development` branch
- [x] uses the `999-SNAPSHOT` version of Quarkus
- [x] has tests (`mvn clean test`)
- [x] works in native (`mvn clean package -Pnative`)
- [x] has integration/native tests (`mvn clean verify -Pnative`)
- [x] makes sure the associated guide must not be updated
- [ ] links the guide update pull request (if needed)
- [x] updates or creates the `README.md` file (with build and run instructions)
- [ ] for new quickstart, is located in the directory _component-quickstart_
- [ ] for new quickstart, is added to the root `pom.xml` and `README.md`

### CODE REVIEW

File: optaplanner-quickstart/pom.xml [hunk 1]
```diff
@@ -10,7 +10,7 @@
     <quarkus.platform.artifact-id>quarkus-bom</quarkus.platform.artifact-id>
     <quarkus.platform.group-id>io.quarkus</quarkus.platform.group-id>
     <quarkus.platform.version>999-SNAPSHOT</quarkus.platform.version>
-    <optaplanner-quarkus.version>8.27.0.Final</optaplanner-quarkus.version>
+    <optaplanner-quarkus.version>8.28.0.Final</optaplanner-quarkus.version>
```
Comment (triceo) [inline optaplanner-quickstart/pom.xml:13]:
```suggestion
    <optaplanner-quarkus.version>8.29.0.Final</optaplanner-quarkus.version>
```

**Review by cescoffier — APPROVED** [review]
Can you squash the commits?

**Comment by triceo** [comment]
@cescoffier It seems like the test failure is not related to my changes. Anything I could/should do about it?

**Review by gsmet — APPROVED** [review]

**Comment by triceo** [comment]
@gsmet I see the opentelementry quickstart failing. I do not understand how my changes could be related to this failure, and I have not seen it fail locally.

**Pull request merged**